Label `BOTTLEROCKET-DATA` at runtime (#2807)
bottlerocket-os/bottlerocket: variants/README.md

Though we've done so here for sake of demonstration, resizing the "data" disk image isn't necessary as it expands to fill the disk on boot.
`partition-plan` is the strategy used for image partitioning, with the options being "split" (the default) or "unified".
The "split" partition strategy has separate volumes for "OS" and "data", while "unified" has "OS" and "data" on a single volume.

All `aws` variants use the "split" partition plan and the published AMIs all have two EBS volumes.

> The root volume contains the "OS" image and the data volume contains the "data" image.

Reviewer: This documentation note makes sense but I wonder if it'll be read by people who need to see it. It's also a bit out of place since it discusses AWS specifics in the general variant feature doc.

It might be better to write this up as an announcement in the discussions area, as we did for the SELinux related changes.

> Important note: Under the "split" partition plan, if the root volume/disk size is increased beyond `os-image-size-gib`, Bottlerocket, at first boot, will create an alternative data partition on the root volume.

Reviewer: ```suggestion
**Important note:** Under the "split" partition plan, if the root volume/disk size is increased beyond `os-image-size-gib`, Bottlerocket, at first boot, will also extend the data partition on the root volume.
```

> This is a preventative measure to allow boot to continue in cases where the data volume is unavailable.

Reviewer: ```suggestion
This allows boot to continue in cases where the data volume is intentionally not attached, and where the goal is to use the additional space on the root volume for local storage.
```

> If the data volume is attached at the time, then there are now two viable data partitions.
> At this point it becomes non-deterministic which data partition Bottlerocket will mount and use.

Reviewer: ```suggestion
However, if the data volume is also attached during first boot, then there will be two viable data partitions, and it is **non-deterministic, undefined behavior** which data partition will actually be used for local storage.
```

> Therefore, it is advisable to _not_ increase the root volume size beyond the default unless the data volume is expected to be unavailable at boot.

Reviewer: ```suggestion
Therefore, it is strongly advised to _not_ increase the root volume size beyond the default unless the data volume is guaranteed to be unavailable at boot.
```
